restli-tools/src/main/java/com/linkedin/restli/tools/clientgen/fluentspec/ComplexKeySpec.java

```diff
@@ -41,14 +42,17 @@ public ComplexKeySpec(String keyKeyType, String paramKeyType, BaseResourceSpec b
     this._paramKeyClassName = _baseResourceSpec.getJavaBindTypeName(paramKeyType);
   }
 
-  public String getParameterizedSignature(Set<String> importNameConflict)
+  public String getParameterizedSignature(Map<String, String> importNameConflict)
   {
     if (_parameterizedSignature == null)
     {
       String keyKeyClassNameShort = ClassUtils.getShortClassName(_keyKeyClassName);
       String paramKeyClassNameShort = ClassUtils.getShortClassName(_paramKeyClassName);
-      _parameterizedSignature = String.format("ComplexResourceKey<%s, %s>", importNameConflict.contains(keyKeyClassNameShort)? _keyKeyClassName: keyKeyClassNameShort,
-          importNameConflict.contains(paramKeyClassNameShort)? _paramKeyClassName:paramKeyClassNameShort);
+      _parameterizedSignature = String.format("ComplexResourceKey<%s, %s>",
+          SpecUtils.checkIfShortNameConflictWithImports(importNameConflict, keyKeyClassNameShort, _keyKeyClassName)?
+           _keyKeyClassName: keyKeyClassNameShort,
+          SpecUtils.checkIfShortNameConflictWithImports(importNameConflict, keyKeyClassNameShort, _keyKeyClassName)?
+          _paramKeyClassName:paramKeyClassNameShort);
```

Comment:
If you are using the short name, then you should also update the map to avoid future conflicts?

Reply:
update the `checkIfShortNameConflictWithImports` so it has side effect for updating